File: src/adventOfCode2020/aoc2020-05.js
const fs = require('fs');

const lines = fs.readFileSync('aoc2020-05.txt', 'utf-8').split(/\r?\n/);
const nbs = lines.map(b => parseInt(b.replace(/[FL]/g, '0').replace(/[BR]/g, '1'), 2)).sort((a, b) => +a - +b);

const part1 = () => nbs[nbs.length - 1];
const part2 = () => nbs.find((n, i) => i > 0 && n !== nbs[i-1] + 1) - 1;

console.log(part1());
console.log(part2());

File: src/adventOfCode2020/aoc2020-05.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn main() -> Result<(), std::io::Error> {
    let lines = read_lines("./src/adventOfCode2020/aoc2020-05.txt")?;
    let mut values = lines
        .flatten()
        .map(|a| a.chars().map(|c| match c {
            'F'|'L' => '0',
            'B'|'R' => '1',
            _ => c }).collect::<String>())
        .flat_map(|c| i32::from_str_radix(&c, 2))
        .collect::<Vec<i32>>();
    values.sort();
    let values2 = values.iter().skip(1);
    println!("Biggest value {}", values.last().unwrap());
    let value3 = values.iter().zip(values2).find_map(|(&a, &b)| if a+1 != b {Some(a+1)} else {None});
    println!("First non consecutive value {}", value3.unwrap());
    Ok(())
}

// The output is wrapped in a Result to allow matching on errors
// Returns an Iterator to the Reader of the lines of the file.
fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
    where P: AsRef<Path>, {
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}